Tests/UnitTests/OfflineEntitlements/OfflineCustomerInfoCreatorTests.swift

```diff
@@ -67,6 +94,42 @@ class OfflineCustomerInfoCreatorTests: TestCase {
             expect(self.mockDiagnosticsTracker.trackedEnteredOfflineEntitlementsModeCalls.value) == 0
         }
     }
+
+    func testTrackErrorEnteringOfflineEntitlementsModeCalledWhenFetcherFails() async throws {
+        self.mockProductEntitlementMappingFetcher.stubbedResult = .init(
+            entitlementsByProduct: ["product": ["entitlement"]]
+        )
+        let error = PurchasedProductsFetcher.Error.foundConsumablePurchase
+        self.mockPurchasedProductsFetcher.stubbedResult = .failure(error)
+
+        do {
+            _ = try await self.creator.create(for: "user")
+            fail("Expected error")
+        } catch {
+            expect(self.mockDiagnosticsTracker.trackedErrorEnteringOfflineEntitlementsModeCalls.value.count) == 1
+            let (reason, msg) = self.mockDiagnosticsTracker.trackedErrorEnteringOfflineEntitlementsModeCalls.value[0]
+            expect(reason) == "one_time_purchase_found"
+            expect(msg) == Strings.offlineEntitlements
+                .computing_offline_customer_info_for_consumable_product.description
```

Comment:
I'm aware that with the current implementation, this check is a bit too strong in the sense that it depends, not only on the implementation of `OfflineCustomerInfoCreatorTests`, but also on the implementation of `PurchasedProductsFetcher.Error`. But I think it's worth it for now to always have a consistent diagnostics report